src/vmm/src/builder.rs

```diff
@@ -528,6 +528,8 @@ pub fn build_microvm_from_snapshot(
         MMIODeviceManager::restore(mmio_ctor_args, &microvm_state.device_states)
             .map_err(MicrovmStateError::RestoreDevices)
             .map_err(RestoreMicrovmState)?;
+    vmm.emulate_serial_init()
+        .map_err(|_| StartMicrovmError::Internal(Error::Serial(std::io::Error::last_os_error())))?;
```

Comment:
why do we disregard the error here? we should propagate it instead of calling `last_os_error` twice